src/app/codegen-data-model-provider/CodegenDataModelProvider.cpp

```diff
@@ -193,6 +192,23 @@ DataModel::ClusterEntry FirstServerClusterEntry(EndpointId endpointId, const Emb
     return DataModel::ClusterEntry::kInvalid;
 }
 
+ClusterId FirstClientClusterId(const EmberAfEndpointType * endpoint, unsigned start_index, unsigned & found_index)
+{
+    for (unsigned cluster_idx = start_index; cluster_idx < endpoint->clusterCount; cluster_idx++)
+    {
+        const EmberAfCluster & cluster = endpoint->cluster[cluster_idx];
+        if ((cluster.mask & CLUSTER_MASK_CLIENT) == 0)
```

Comment:
Please use the relevant accessors on EmberAfCluster instead of poking the mask directly.

Reply:
Added a `IsClient()` functions to `EmberAfCluster` struct and used it here.